src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -243,7 +243,10 @@ public override BoundNode VisitLocalFunctionStatement(BoundLocalFunctionStatemen
             bool hasConstraintsWithNullableReferenceTypes = typeParameters.Any(
                typeParameter => typeParameter.ConstraintTypesNoUseSiteDiagnostics.Any(
                    typeConstraint => typeConstraint.ContainsNullableReferenceTypes()));
-            if (hasConstraintsWithNullableReferenceTypes)
+            bool hasReturnTypeWithNullableReferenceTypes = node.Symbol.ReturnType.ContainsNullableReferenceTypes();
```

Comment:
:memo: I'd missed this scenario in my previous PR. We need to register uses of `?` annotations in parameters and return type of local functions as well.